### DISCUSSION TITLE
Does vllm support CPU?

### DISCUSSION THREAD

**Discussion by gokul427** [vllm-project/vllm#999] [Q&A]
Can we use vllm only on CPU without GPU machine?

**Comment by mspronesti** [comment]
I think the short answer is no, as vLLM's engine relies on custom kernels written in CUDA.

**Comment by BrightXiaoHan** [comment]
You can try ctranslate2 or llama.cpp.

**Reply by VpkPrasanna** [reply]
obviously i know this library has been designed on top of cuda kernels for better GPU utilization  , but the inference will pretty slow right on CPU? is there any workaround to load these quantized models on CPU to server faster ?
#2326

**Comment by hughesadam87** [comment]
This was not clear to me either - any way to highlight this in bold somewhere on main docs?  Sorry if I overlooked.  

I am trying to do some local testing - that's my use case

**Comment by zengqingfu1442** [comment]
Intel CPU is supported from my own experience.

**Reply by theBeginner86** [reply]
+1

**Reply by theBeginner86** [reply]
Reference: https://docs.vllm.ai/en/stable/getting_started/installation/cpu.html
